src/PackageProject.cmake

```diff
@@ -203,3 +203,45 @@ function(package_project)
 
   include("${_ycm_SOURCE_DIR}/modules/AddUninstallTarget.cmake")
 endfunction()
+
+# A function that includes ${CMAKE_CURRENT_SOURCE_DIR}/include as the header directory of the target.
+# A variable `<target_name>_HEADER_DIRECTORY` will be created to represent the header directory path.
+function(target_include_header_directory target)
```

Comment:
Maybe we should call this `target_include_public_directory`?

Reply:
The word `public` might lead to misunderstand as header-only library actually sets INTERFACE include directory.

Reply:
Though still a poor name, I'd like to call `target_include_interface_directory` instead of `target_include_public_directory` as  `PUBLIC` and `INTERFACE` items will both populate the `INTERFACE_INCLUDE_DIRECTORIES` property.

Reply:
Though still a poor name, I'd like to call `target_include_interface_directory` instead of `target_include_public_directory` as  `PUBLIC` and `INTERFACE` items will both populate the `INTERFACE_INCLUDE_DIRECTORIES` property.

Reply:
Though still a poor name, I'd like to call `target_include_interface_directory` instead of `target_include_public_directory` as  `PUBLIC` and `INTERFACE` items will both populate the `INTERFACE_INCLUDE_DIRECTORIES` property.